### PR TITLE
fix: drag and drop the event List

### LINKED ISSUE #13067: Drag Event Currently It Having A Button Which Is Little Bit Bad For UX
### Is your proposal related to a problem?

https://www.loom.com/share/77bde77b9bf54992b4baf270fd2c3b65?sid=e2405963-2bba-4dba-8150-430c113c04c6

We Can Do Drag Option Like By Holding On It Put Where He Want

### Describe the solution you'd like

implement a DraggableList component that uses react-dnd for drag-and-drop functionality.

(Write your answer here.)

### Additional context

(Write your answer here.)

### Requirement/Document

Yes  https://react-dnd.github.io/react-dnd/docs/tutorial Here Is Documentation By Where We Can Implement it

---
##### House rules
- If this issue has a `🚨 needs approval` label, don't start coding yet. Wait until a core member approves feature request by removing this label, then you can start coding.
  - For clarity: Non-core member issues automatically get the `🚨 needs approval` label.
  - Your feature ideas are invaluable to us! However, they undergo review to ensure alignment with the product's direction.

### PR DESCRIPTION
**Pull request by Airbornharsh**
## What does this PR do?

Fixes #13067 

https://github.com/calcom/cal.com/assets/8019099/3c2a53f5-dfc7-438a-98d4-3b5dee66f2e2

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Tests (Unit/Integration/E2E or any other test)
- [ ] This change requires a documentation update

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: apps/web/pages/event-types/index.tsx [hunk 1]
```diff
@@ -396,7 +424,12 @@ export const EventTypeList = ({
           const isChildrenManagedEventType =
             type.metadata?.managedEventConfig !== undefined && type.schedulingType !== SchedulingType.MANAGED;
           return (
-            <li key={type.id}>
+            <li
+              key={type.id}
+              draggable={!readOnly}
+              onDragStart={(e) => dragStart(e, index)}
+              onDragEnter={() => dragOver(index)}
+              onDragEnd={drop}>
```
Comment (PeerRich) [inline apps/web/pages/event-types/index.tsx:432]: i wonder if we put this on the up and down buttons:

CleanShot 2024-01-07 at 10 39 28@2x


and give it class="pointer-grab"

**Comment by PeerRich** [comment]
also there is a slight delay after dropping where the container goes back to its old position and then moves to the correct position

File: apps/web/pages/event-types/index.tsx (hunk 1 above)
Comment (Airbornharsh) [reply apps/web/pages/event-types/index.tsx:432]: Ok I will add

**Comment by Airbornharsh** [comment]
@PeerRich 

https://github.com/calcom/cal.com/assets/95279849/ca1638e6-6974-48f1-a45f-50a5f3ff0b55

Added  Drag on the Element and there was no delay in Dragging

**Comment by PeerRich** [comment]
I was referring on adding the grab event only on the up and down buttons:

`<ArrowButton onClick={() => moveEventType(index, 1)} arrowDirection="down" />`

**Comment by Airbornharsh** [comment]
@PeerRich OK I will Update it

**Comment by PeerRich** [comment]
@Airbornharsh let me know if you need help

**Comment by Airbornharsh** [comment]
@PeerRich I have Added Drag in the arrow, you can Review it

**Comment by Udit-takkar** [comment]
@PeerRich the issue has been marked "not planned". do we need this?

**Comment by keithwillcode** [comment]
Closing based on @PeerRich comments on the referenced issue.

**Pull request closed without merging**